kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/Either.java

```diff
@@ -46,6 +46,6 @@ public boolean isRight() {
     
     public <X> X cata(Function<L,X> left, Function<R,X> right) {
         // warning: left.invoke, because of FEEL specs, could return null. The below is the safest way to implement cata over this Either.
-        return isLeft() ? left.apply( this.left.get() ) : right.apply( this.right.get() );
+        return isLeft() ? left.apply( this.left.get() ) : right.apply( this.right.orElse( null ) );
```

Comment:
Yep, can you only kindly add a comment to `Either.isRight()` just before merging, as a note for posterity that:
this implementation of Either is tailored to support FEEL spec as functions could explicitly return null, hence two definition of isRight are possible

- negated of `isLeft`, where `isLeft()` is defined as `left.isPresent()`
or
- `isRight` defined as `right.isPresent() || (!right.isPresent() && !left.isPresent())`

so we opted for the former as a matter of simplicity?

I don't want that later I may inadvertently touch this Either making troubles down the line :)